lua/refactoring/tests/refactor_spec.lua

```diff
@@ -15,6 +15,7 @@ local extension_to_filetype = {
 local refactor_id_to_name = {
     ["119"] = "extract_var",
     ["106"] = "extract",
+    ["123"] = "inline_var",
```

Comment:
now that I think about it.  Perhaps we should have some "meta" data for each type of test. 

I think this gets us to the next goal post, but it doesn't really solve the issue much.  What if we figure out a way for the test itself to be able to report "required" features.  (LSP, TS, perhaps even specific TS query features, language features, code generation features).  

Thoughts?

Reply:
I like this  approach 100x better. Refactoring PR to now have all refactor functions in same method and metadata for `LSP` if it requires it. Also checks if there is an input file to run it (`inline_var` does not require input file). Also moved some stuff to `test_utils` that's used across other tests. Let me know if you like this approach more.